esrally/driver/driver.py

```diff
@@ -696,6 +707,12 @@ def prepare_benchmark(self, t):
         else:
             self.wait_for_rest_api(es_clients)
             self.target.cluster_details = self.retrieve_cluster_info(es_clients)
+            serverless_mode = self.config.opts("driver", "serverless.mode", default_value=False, mandatory=False)
+            serverless_operator = self.config.opts("driver", "serverless.operator", default_value=False, mandatory=False)
+            if serverless_mode and serverless_operator:
+                build_hash = self.retrieve_build_hash_from_nodes_info(es_clients)
+                self.logger.info("Retrieved actual build hash [%s] from serverless cluster.", build_hash)
+                self.target.cluster_details["version"]["build_hash"] = build_hash
```

Comment:
What about the stats in the metric store?

There is some code in https://github.com/elastic/rally/blob/c3b04f4d97de7c379bd78905d83df8bc71c7d229/esrally/telemetry.py#L1778-L1781 and in my tests it results in

![image](https://github.com/elastic/rally/assets/1754575/2c4c9a13-a749-4957-8c7b-930a31f6c033)

(correctly)

but 

![image](https://github.com/elastic/rally/assets/1754575/d0ece4ac-22e8-47a0-89f6-58a9ae467248)

whereas I see in the logs `Retrieved actual build hash [ad5c80e1b49f10b42d438253d7cc0b9753c156b9] from serverless cluster.`

Additionally, shouldn't it also be collected by the [node stats telemetry device](https://github.com/elastic/rally/blob/c3b04f4d97de7c379bd78905d83df8bc71c7d229/esrally/telemetry.py#L779-L842)?

Reply:
Telemetry device modifications are scoped separately in https://elasticco.atlassian.net/browse/ES-6459. After discussion with @dliappis we thought the necessary changes to override build hash in `ClusterEnvironmentInfo` are small, so I went ahead and added [da34a7d](https://github.com/elastic/rally/pull/1756/commits/da34a7d843237e1c387e081b3949f37e31afe9fb). PTAL.